packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,199 @@
+                }
+              },
+            ),
+          ),
+        ),
+        if (widget.showStatus) ...[
+          if (widget.username != null) ...[
+            ResultBuilder<NextcloudUserStatusPublicStatus?>(
+              stream: _userStatusBloc.otherUserStatuses.map(
+                (final statuses) => statuses[effectiveUsername] ?? Result.loading(),
+              ),
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ] else ...[
+            ResultBuilder<NextcloudUserStatusStatus?>(
+              stream: _userStatusBloc.userStatus,
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ],
+        ],
+      ],
+    );
+  }
+
+  Widget _userStatusIcon({
+    required final bool loading,
+    required final Object? error,
+    required final _UserStatus? userStatus,
+  }) {
+    final hasEmoji = userStatus?.icon != null;
+    final factor = hasEmoji ? 2 : 3;
+    return SizedBox(
+      height: widget.size,
+      width: widget.size,
+      child: Align(
+        alignment: Alignment.bottomRight,
+        child: Container(
+          height: widget.size / factor,
+          width: widget.size / factor,
+          decoration: loading || error != null || userStatus == null || hasEmoji
+              ? null
+              : BoxDecoration(
+                  shape: BoxShape.circle,
+                  color: _userStatusToColor(userStatus.status),
+                ),
+          child: loading
+              ? CircularProgressIndicator(
+                  strokeWidth: 1.5,
+                  color: Theme.of(context).colorScheme.onPrimary,
+                )
+              : error != null && (error is! NextcloudApiException || error.statusCode != 404)
+                  ? Icon(
+                      Icons.error_outline,
+                      size: widget.size / factor,
+                      color: Colors.red,
+                    )
+                  : hasEmoji
+                      ? Text(
+                          userStatus!.icon!,
+                          style: const TextStyle(
+                            fontSize: 16,
+                          ),
+                        )
+                      : null,
```

Comment:
Also quite hard to read.

Maybe have an external `child` variable before the return statement that gets conditionally assigned.